### DISCUSSION TITLE
Saving issue re. UNICODE

### DISCUSSION THREAD

**Discussion by NWermes** [WinEdt-Team/WinEdt-Support#12] [1.1 General Help and Discussion]
Hi, when saving in WinEdt I sometimes encounter the following comment
"Some characters were lost when converting from UNICODE to CP 0"
Is there a way to find out which characters cause this comment?
Thanks

**Comment by Alex-Simonic** [comment]
This is not a WinEdt bug or issue. WinEdt is a Unicode editor and by default, it saves files in UTF8 format (as it should). You have chosen to change this and save files in the old Windows CP0 code page which can only accommodate 256 characters. If you type (or more likely paste) text that contains characters outside the CP0 range they will be replaced by a question mark (not much else could be done -- Pigeonhole principle explains why). WinEdt will warn you about lost characters but CP0 choice (instead of UTF8) is yours and so are the consequences.

Now to answer your question:

1. You can use RegEx Search for \u to find Unicode Characters (Help explains).

2. In the View menu you can enable Unicode Preview which will display such characters with their code (e.g. π = U+03C0). This is particularly useful because some Unicode characters can be hard to spot and may look like ordinary ones (including spaces, quotes, etc...).

**Reply by NWermes** [reply]
Thank you very much, indeed.

**Reply by NWermes** [reply]
I did not choose Cp0 advertently. Maybe it is because I use a German keyboard and write German text? Can I change it back to UTF8?

**Reply by Alex-Simonic** [reply]
If you did not change WinEdt's default UTF8 settings then there is another explanation:

Your document was created with some other software (or a very old version of WinEdt) that does not use UTF8 and your document already contains some 8-bit characters (like German umlauts). WinEdt automatically detects that the document is not in UTF8 and it loads it in your default code page. The CP0 format is preserved when you save the document (as it should be in order to allow WinEdt to work with legacy documents).

Start the Document Settings Dialog (in WinEdt's Document Menu), go to the last page (CP Converter), change the document format from ANSI to UTF-8 in the drop-down list, and use Apply and Save to apply the change.

But before you do so you should understand that this may require changes to your preamble (or a different TeX engine) or else TeX will fail to compile it. Also, if your TeX document contains subdocuments they all need to use the same encoding (format). UTF8 is definitely the way to go for those who understand the issue. But for old legacy documents written in (and for) ANSI code pages, the conversion may not be worth the trouble. That's why WinEdt allows you to work in ANSI format (as long as you don't paste in something that cannot be translated into a single code page).

German keyboard is certainly not a problem. UTF8 can handle 64,000+ characters (as opposed to 256 in a single code page) and it is used for mixed or complex languages, etc... But this may also require a Unicode TeX engine (XeTeX or LuaTeX) rather than pdfTeX. 

This is not a WinEdt-issue... WinEdt can handle both and the choice (and the consequence) is yours.

Hope this explains it!

**Reply by NWermes** [reply]
Yes, this explains everything I wanted to know in detail. Many thanks.
I am now a bit afraid, though, to switch to UTF8 as I am working for many years on various editions of a book (1000 pages) with
figures and style-file modifications.
